src/components/rating/Rating.jsx

```diff
@@ -91,30 +104,33 @@ class Rating extends Component {
         <div className="sg-rate-box__rate">
           {rateString}
         </div>
-        <div className="sg-rate-box__stars-container" onMouseEnter={this.onMouseEnter} onMouseLeave={this.onMouseLeave}>
+        <div className="sg-rate-box__stars-container" onMouseLeave={this.onMouseLeave}>
           <div className="sg-rate-box__filled-stars" style={{width: `${100 * rate / metricSize}%`}}>
             {starsProps.map(props => <Star key={props.key} {...props} />)}
           </div>
           <div className="sg-rate-box__background-stars">
-            {starsProps.map(props => <Star key={props.key} {...props} />)}
+            {starsProps.map(props =>
+              <Star key={props.key} onMouseEnter={this.starsMouseEnterFunctions[props.key]} {...props} />)}
```

Comment:
why onEnter is here and onLeave is on the container?

Reply:
There's no real need for listening on Star 'mouseleave' event, when you are switching from star to star it's easier to check for update from onEnter than on both leave and enter listeners.

On the other side, its important to know when you are actually leaving whole component.
Generally these two behaviours allow to cover all current use cases:
- hovering stars - entering star equals to entering component,
- hovering stars - going from star 1 to star 2 (no onLeave ==> less events triggered)
- leaving component

Reply:
hm, makes sense but maybe we should be more explicit? e.g. `onLeaveComponent`, `onEnterStar`? we may need `onEnterComponent` and `onLeaveStar` in the future.

BTW `onLeaveComponent` isn't the same as `<Rating onMouseLeave=` ?